mesonbuild/dependencies/base.py

```diff
@@ -390,24 +391,21 @@ def find_config(self, versions=None):
         if not isinstance(versions, list) and versions is not None:
             versions = listify(versions)
 
-        if self.env.is_cross_build() and not self.native:
-            cross_file = self.env.cross_info.config['binaries']
-            try:
-                tools = [cross_file[self.tool_name]]
-            except KeyError:
+        for_machine = MachineChoice.BUILD if self.native else MachineChoice.HOST
+        tool = self.env.binaries[for_machine].lookup_entry(self.tool_name)
+        if tool is not None:
+            tools = [tool]
+        else:
+            if self.env.is_cross_build() and not self.native:
                 mlog.warning('No entry for {0} specified in your cross file. '
                              'Falling back to searching PATH. This may find a '
                              'native version of {0}!'.format(self.tool_name))
-                tools = self.tools
-        elif self.tool_name in self.env.config_info.binaries:
-            tools = [self.env.config_info.binaries[self.tool_name]]
-        else:
-            tools = self.tools
+            tools = map(lambda tool: [tool], self.tools)
```

Comment:
This is made a list to match the full generality of cross file `[binaries]` entries